packages/astro/src/core/build/generate.ts

```diff
@@ -259,7 +256,8 @@ async function generatePage(
 	// prepare information we need
 	const { config, internals, logger } = pipeline;
 	const pageModulePromise = ssrEntry.page;
-	const pageInfo = getPageDataByComponent(internals, pageData.route.component);
+	// TODO: Why do we need to get PageData from PageData?
+	const pageInfo = getPageData(internals, pageData.route.component, pageData.route.route);
```

Comment:
Is it the same exact object?

Reply:
@matthewp Hum...

First, it looks like it's the same exact object.

Secondly, this code is broken haha `getPageDataByComponent(internals, pageData.route.component);` will always return `undefined`.

Luckily, It looks like `pageInfo` is only used by `const scripts = pageInfo?.hoistedScript ?? null;`, and then `scripts` is gived to : 
```ts
const generationOptions: Readonly<GeneratePathOptions> = {
    pageData,
    linkIds,
    scripts,
    styles,
    mod: pageModule,
};
```

And then never used anywhere? Tbh, none of those params (except `mod`) seems used anywhere... Legacy code?